libraries/standard/mqtt/src/iot_mqtt_serialize.c

```diff
@@ -487,30 +488,30 @@ static bool _connectPacketSize( const IotMqttConnectInfo_t * pConnectInfo,
     if( pConnectInfo->awsIotMqttMode == true )
     {
         #if AWS_IOT_MQTT_ENABLE_METRICS == 1
-            connectPacketSize += AWS_IOT_METRICS_USERNAME_LENGTH + sizeof( uint16_t );
+            connectPacketSize += AWS_IOT_METRICS_USERNAME_LENGTH +
+                                 pConnectInfo->userNameLength + sizeof( uint16_t );
+            encodedUserName = true;
         #endif
     }
+
```

Comment:
Can the SDK metrics and username for authentication coexist in the same MQTT Connect message?  If so what is the layout?  Which comes first and how much space is reserved for each?  If this is the case, does the maximum username length check on line 135 need to be adjusted?

Reply:
Yes, good catch, thank you. I've updated our length checks to reflect that metrics + logon name may be combined.